src/main/kotlin/minesweeper/domain/MineSweeperBoardFactory.kt

```diff
@@ -0,0 +1,35 @@
+package minesweeper.domain
+
+fun factory(block: MineSweeperBoardBuilder.() -> Unit): MineSweeperBoard {
+    return MineSweeperBoardBuilder().apply(block).build()
+}
+
+class MineSweeperBoardBuilder {
+    private lateinit var height: Height
+    private lateinit var width: Width
+    private lateinit var countOfMine: CountOfMine
+
+    fun height(value: Int) {
+        height = Height(value)
+    }
+
+    fun width(value: Int) {
+        width = Width(value)
+    }
+
+    fun countOfMine(value: Int) {
+        countOfMine = CountOfMine(value)
+    }
+
+    fun build(): MineSweeperBoard {
+        val zoneArea = height.value * width.value
+        require(zoneArea >= countOfMine.value) { "지뢰 갯수는 총 면적보다 클 수 없습니다. 총 면적 = $zoneArea, 지뢰 갯수 = $countOfMine" }
+        val safeZones = List(zoneArea - countOfMine.value) { Zone(false) }
+        val mineZones = List(countOfMine.value) { Zone(true) }
```

Comment:
```suggestion
        val safeZones = List(zoneArea - countOfMine.value) { SafeZone() }
        val mineZones = List(countOfMine.value) { MineZone() }
```
SafeZone에 대해서는 SafeZone으로, MineZone에 대해서는 MineZone으로 만드는건 어떨까요?

Reply:
이 부분도 앞서 언급했던 'KISS 원칙'을 생각한 것인데, 지금 당장은 `지뢰존`과 `안전존`을 나누는 의미가 크게 없다고 생각되서 단순히 프로퍼티로 `mine`을 가지고 있는지 여부로 체크를 했습니다. 😃

2단계에서는 피드백 주신대로 별도의 클래스로 분리해보겠습니다. 👍